src/App.vue

```diff
@@ -560,16 +556,16 @@ export default {
 
       // store business info
       this.setIdentifier(this.tempRegNumber)
-      this.setEntityType(legalType)
+      this.setLegalType(legalType)
 
       // Draft Applications are always in good standing
       this.setGoodStanding(true)
 
-      // store NR Number if present
+      // save local NR Number if present
       this.localNrNumber = nameRequest.nrNumber || null
 
       // store Legal Name if present
-      this.setEntityName(nameRequest.legalName || null)
+      if (nameRequest.legalName) this.setLegalName(nameRequest.legalName)
```

Comment:
I think the new code is better -- leave the initial store value alone if we don't set an entity name here.